Is there support for using with `System.import()` (or alternative?)

I am using `System.import()` in my React project for lazy loading components and data to reduce initial bundle size.

```javascript
["module-resolver", {"alias": { "data": "./src/data/"}}]
```

Sample code
```javascript
System.import('data/faqs')
  .then(response => response.default)
  .then(faqs => this.setState({faqs}))
```

Error
```shell
ERROR in ./src/scripts/app/pages/FAQ.js
Module not found: Error: Can't resolve 'data/faqs' in '<truncated-path>/src/scripts/app/pages'
 @ ./src/scripts/app/pages/FAQ.js 42:6-39
 @ ./src/scripts/bundles/app.js
 @ multi app
```

Regular imports (`import faqs from 'data/faqs'`) are working as expected, but not when using `System.import()`

If `System.import()` is not supported is there an alternative I can use?
For now I am just creating aliases within webpack config for the affected imports like below

```javascript
resolve: {
  alias: {
    data: path.join(__dirname, 'src/data/')
  }
}
```

Would be nice if babel could handle all the aliases if possible :)

--- a/src/index.js
+++ b/src/index.js
@@ -148,6 +148,33 @@ export default ({ types: t }) => {
         }
     }
 
+    function transformSystemImportCall(nodePath, state, cwd) {
+        const calleePath = nodePath.get('callee');
+
+        if (!(
+            t.isMemberExpression(calleePath.node) &&
+            t.isIdentifier(calleePath.node.object, { name: 'System' }) &&
+            t.isIdentifier(calleePath.node.property, { name: 'import' })
+        )) {
+            return;
+        }
+
+        const args = nodePath.get('arguments');
+        if (!args.length) {
+            return;
+        }
+
+        const moduleArg = args[0];
+        if (moduleArg.node.type === 'StringLiteral') {
+            const modulePath = mapModule(moduleArg.node.value, state.file.opts.filename, state.opts, cwd);
+            if (modulePath) {
+                nodePath.replaceWith(t.callExpression(
+                    calleePath.node, [t.stringLiteral(modulePath)],
+                ));
+            }
+        }
+    }
+
     return {
         manipulateOptions(babelOptions) {
             const findPluginOptions = babelOptions.plugins.find(plugin => plugin[0] === this)[1];
@@ -188,6 +215,7 @@ export default ({ types: t }) => {
 
                     transformRequireCall(nodePath, state, this.moduleResolverCWD);
                     transformJestCalls(nodePath, state, this.moduleResolverCWD);
+                    transformSystemImportCall(nodePath, state, this.moduleResolverCWD);
 
                     // eslint-disable-next-line no-param-reassign
                     nodePath.node.seen = true;

--- a/test/system.test.js
+++ b/test/system.test.js
@@ -1,0 +1,41 @@
+/* eslint-env jest */
+import { transform } from 'babel-core'; // eslint-disable-line import/no-extraneous-dependencies
+import plugin from '../src';
+
+describe('System.import', () => {
+    const transformerOpts = {
+        babelrc: false,
+        plugins: [
+            [plugin, {
+                root: [
+                    './test/examples/components',
+                    './test/examples/foo',
+                ],
+                alias: {
+                    utils: './src/mylib/subfolder/utils',
+                },
+            }],
+        ],
+    };
+
+    it('should resolve the path based on the root config', () => {
+        const code = 'System.import("c1").then(() => {}).catch(() => {});';
+        const result = transform(code, transformerOpts);
+
+        expect(result.code).toBe('System.import("./test/examples/components/c1").then(() => {}).catch(() => {});');
+    });
+
+    it('should alias the path', () => {
+        const code = 'System.import("utils").then(() => {}).catch(() => {});';
+        const result = transform(code, transformerOpts);
+
+        expect(result.code).toBe('System.import("./src/mylib/subfolder/utils").then(() => {}).catch(() => {});');
+    });
+
+    it('should not change the path', () => {
+        const code = 'System.import("./utils").then(() => {}).catch(() => {});';
+        const result = transform(code, transformerOpts);
+
+        expect(result.code).toBe('System.import("./utils").then(() => {}).catch(() => {});');
+    });
+});

